ios/FluentUI/AvatarGroup/AvatarGroup.swift

```diff
@@ -185,48 +186,92 @@ public struct AvatarGroup: View {
+                            let rtlNoRingPaddingInterspace = nextAvatarHasRing ? -x - ringOffset - ringGapOffset : -x - ringOuterGap
+                            let stackPadding = currentAvatarHasRing ? ringPaddingInterspace : noRingPaddingInterspace
+
+                            let xPosition = currentAvatarHasRing ? x - ringGapOffset : x - ringOuterGap
+                            let xPositionRTL = currentAvatarHasRing ? rtlRingPaddingInterspace : rtlNoRingPaddingInterspace
+                            let xOrigin = Locale.current.isRightToLeftLayoutDirection() ? xPositionRTL : xPosition
+                            let yOrigin = sizeDiff / 2
+                            let cutoutSize = isLastDisplayed ? ringGapOffset + imageSize : nextAvatarSize
+
+                            VStack {
+                                avatarView
+                                    .transition(.identity)
+                                    .modifyIf(needsCutout, { view in
+                                        view.mask(Avatar.AvatarCutout(
+                                            xOrigin: xOrigin,
+                                            yOrigin: yOrigin,
+                                            cutoutSize: cutoutSize)
+                                                    .fill(style: FillStyle(eoFill: true)))
+                                    })
+                            }
+                            .padding(.trailing, tokens.style == .stack ? stackPadding : interspace)
+                            .animation(Animation.linear(duration: 0.1))
+                            .transition(AnyTransition.move(edge: .leading))
+                        }
+                        if overflowCount > 0 {
+                            VStack {
+                                createOverflow(count: overflowCount)
+                            }
+                            .transition(AnyTransition.move(edge: .leading))
+                        }
+                    }
+                    .frame(height: geo.size.height,
+                           alignment: .leading)
+                }
+                .frame(width: calcWidth(hasOverflow: overflowCount > 0), height: groupHeight, alignment: .leading)
             }
         }
+
+        return avatarGroupContent
+    }
+
+    private func calcWidth(hasOverflow: Bool) -> CGFloat {
+        var width: CGFloat = 0
+        let maxShown = state.maxDisplayedAvatars < state.avatars.count ? state.maxDisplayedAvatars : state.avatars.count
+        let interspace: CGFloat = state.style == .stack ? tokens.interspace : tokens.interspace
+        let gapOffset: CGFloat = tokens.ringOuterGap * 2
+        let ringGapOffset: CGFloat = tokens.ringThickness + tokens.ringInnerGap + tokens.ringOuterGap
+        let imageSize: CGFloat = tokens.size.size
+        for index in 0 ..< maxShown {
```

Comment:
is this for loop the best and fastest way to calculate the total width of your avatarGroup? if I understand the intent of this helper function is just find the total width which doesn't necessary need the for loop because you don't need to set the individual frames of each avatar's location.  If I would have pseudo code this wouldn't be something like 

total number of avatar shown X avatar size  + ( ( count -1 /* number of inner gaps */) X gap between avatars /* it can be positive or negative depending on facepile vs face stack */ )

Reply:
I think the problem is that different Avatars can have different sizes, depending on whether they have rings or not.

Reply:
thanks for the explanation! something definitely should be written in comment. Although, I still think we can refactor a bit and share "true" avatar size frame logic between both this function and the other parts of geometricreader code for simplification?